packages/react/src/components/Forms/DatePicker/index.tsx

```diff
@@ -0,0 +1,176 @@
+import { LabelledBox } from ":/components/Forms/LabelledBox";
+import { DateField } from ":/components/Forms/DatePicker/DateField";
+import { Calendar } from ":/components/Forms/DatePicker/Calendar";
+import { Popover } from ":/components/Popover";
+import { useCunningham } from ":/components/Provider";
+
+type DatePickerProps = PropsWithChildren &
+  FieldProps & {
+    label: string;
+    name?: string;
+    value?: null | Date | string;
+    defaultValue?: Date | string;
+    minValue?: Date | string;
+    maxValue?: Date | string;
+    onChange?: (value: string) => void | undefined;
+    disabled?: boolean;
+  };
+
+export const DatePicker = ({
+  label,
+  name,
+  disabled = false,
+  ...props
+}: DatePickerProps) => {
+  if (props.defaultValue && props.value) {
+    throw new Error(
+      "You cannot use both defaultValue and value props on DatePicker component"
+    );
+  }
+
+  const parseCalendarDate = (
+    rawDate: Date | string | undefined
+  ): undefined | CalendarDate => {
+    if (!rawDate) {
+      return undefined;
+    }
+    try {
+      const ISODateString = new Date(rawDate).toISOString();
+      return toCalendarDate(parseAbsoluteToLocal(ISODateString));
+    } catch (e) {
+      throw new Error(
+        "Invalid date format when initializing props on DatePicker component"
+      );
+    }
+  };
+
+  const datePickerOptions: DatePickerStateOptions<DateValue> = {
+    value:
+      // Force clear the component's value when passing null or an empty string.
+      props.value === "" || props.value === null
+        ? null
+        : parseCalendarDate(props.value),
+    defaultValue: parseCalendarDate(props.defaultValue),
+    minValue: parseCalendarDate(props.minValue),
+    maxValue: parseCalendarDate(props.maxValue),
+    onChange: (value: DateValue | null) => {
+      props.onChange?.(value?.toString() || "");
+    },
+    shouldCloseOnSelect: true,
+    granularity: "day",
+    isDisabled: disabled,
+    label,
+  };
+
+  const state = useDatePickerState(datePickerOptions);
+  const pickerRef = useRef<HTMLDivElement>(null);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const { t } = useCunningham();
+
+  const { fieldProps, buttonProps, groupProps, calendarProps } = useDatePicker(
+    datePickerOptions,
+    state,
+    wrapperRef
+  );
+
+  const [labelAsPlaceholder, setLabelAsPlaceholder] = useState(true);
+  const [isDateInvalid, setIsDateInvalid] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    setLabelAsPlaceholder(!state.isOpen && !state.value);
+  }, [state.value, state.isOpen]);
+
+  useEffect(() => {
+    setIsDateInvalid(
+      state.validationState === "invalid" || props.state === "error"
+    );
+  }, [state.validationState, props.state]);
+
+  // onPress props don't exist on the <Button /> component.
+  // Remove it to avoid any warning.
+  const { onPress: onPressToggleButton, ...otherButtonProps } = buttonProps;
```

Comment:
Is it a type issue ?

Reply:
`react-aria` provide a headless implementation of a button component. Their button component hasn't the same props as the native button element. Thus, passing their button props to our component, which uses the native one, triggers a warning. `onPress` or `isDisabled` doesn't exist on native elements. 
I might have miss smth, lemme know.